### DISCUSSION TITLE
How CopyQ pastes data perfectly on respective window textboxes?

### DISCUSSION THREAD

**Discussion by MashukeAlam** [hluk/CopyQ#1662] [Q&A]
When I click on CopyQ items, that item got pasted in textbox of desired window. I am making a similar app using Qt5 which will be a emoji picker. I want to know how CopyQ does this thing. Sorry for bothering, any help will be appreciated. Can anyone tell me the filename and the strategy it uses to paste data? 

Thanks.

**Comment by hluk** [comment]
CopyQ simulates Shift+Insert key press. The code is platform-dependent and may not work in some cases - e.g. on Wayland you would need to use utility like `ydotool` instead.

Most of the code is in `src/platform/*/*platformwindow*` files.
